hmt_escrow/job.py

```diff
@@ -459,19 +447,24 @@ def bulk_payout(self,
 
         Args:
             payouts (List[Tuple[str, int]]): a list of tuples with ethereum addresses and amounts.
-            results (Dict): the final answer results stored by the Reputation Oracle.
+            results (Dict): the repO answer results stored by the Reputation Oracle.
             pub_key (bytes): the public key of the Reputation Oracle.
+            gas (int): Optional, gas limit
+            storeOnchain (bool): Store data onchain. Saves 10k gas
 
         Returns:
             bool: returns True if paying to ethereum addresses and oracles succeeds.
 
         """
-        (hash_, url) = upload(results, pub_key)
+        (hash_, url) = upload(results, pub_key, ipnsKeypairName='reputation-oracle-results-' + self.job_contract.address)
         eth_addrs = [eth_addr for eth_addr, amount in payouts]
         hmt_amounts = [int(amount * 10**18) for eth_addr, amount in payouts]
 
         txn_func = self.job_contract.functions.bulkPayOut
-        func_args = [eth_addrs, hmt_amounts, url, hash_, 1]
+
+        chainUrl  = url   if storeOnchain else ''
```

Comment:
I would modify these camelCases to snake_cases to have a unified way to name variables. More pythonish.